packages/ckeditor5-html-support/src/htmlcomment.ts

```diff
@@ -44,7 +44,7 @@ export default class HtmlComment extends Plugin {
 		editor.conversion.for( 'upcast' ).elementToMarker( {
 			view: '$comment',
 			model: ( viewElement, { writer } ) => {
-				const root = this.editor.model.document.getRoot()!;
+				const root = this.editor.model.document.getRoot( this.editor.model.document.selection.getFirstRange()!.root.rootName! )!;
```

Comment:
This is incorrect. Every HTML-comment-related attribute should be set on the root that contains the comment. `root` here must be a specific model root, the one that holds the comment.

There are three solution candidates for this:

1.  Get the view root from `viewElement` and use `this.editor.editing.mapper` to map view root to model root.
2.  Get the view root from `viewElement` and check if it has `rootName` set correctly. I am not sure that we have a precise rule that respective view and model roots should have the same `rootName`. But that would make sense and if this comes up (i.e. somebody has a problem because their model and view root has different root names) I would recommend fixing the root names.
3.  Change this to "low level" converter and use `modelCursor` value, which is unfortunately not available in this upcast conversion helper. `modelCursor` is a position in model where "conversion takes place", so it would be enough to check `modelCursor.root`. But that would be a bigger change to this part of the code, so I am more inclined to go with solution 2 or 1.